runtime/nativeMain/src/kotlinx/benchmark/NativeBlackhole.kt

```diff
@@ -1,13 +1,37 @@
 package kotlinx.benchmark
 
 actual class Blackhole {
-    actual fun consume(obj: Any?) {}
-    actual fun consume(bool: Boolean) {}
-    actual fun consume(c: Char) {}
-    actual fun consume(b: Byte) {}
-    actual fun consume(s: Short) {}
-    actual fun consume(i: Int) {}
-    actual fun consume(l: Long) {}
-    actual fun consume(f: Float) {}
-    actual fun consume(d: Double) {}
+    @ThreadLocal
+    companion object {
+        var consumer = 0
+    }
+    actual fun consume(obj: Any?) {
+        // hashCode now is implemented as taking address of object, so it's suitable now.
+        // If implementation is changed `Blackhole` should be reimplemented.
+        consumer += obj.hashCode()
+    }
+    actual fun consume(bool: Boolean) {
+        consumer += bool.hashCode()
```

Comment:
Here and below, this is still a write, moreover, it is a write to `@ThreadLocal object` field, so it's unlikely to be `add` instruction to a known address.

Let's please avoid any writes in performance sensitive-code